src/providers/auth.provider.tsx

```diff
@@ -6,17 +6,29 @@ import {
     signInWithEmailAndPassword,
     createUserWithEmailAndPassword,
     GithubAuthProvider,
+    GoogleAuthProvider,
+    signInWithCredential,
     signInWithPopup,
 } from "firebase/auth";
+import { CredentialResponse } from "@react-oauth/google";
 
 export type UserWithRole = User & { hawkAdmin: boolean };
 
+interface GoogleLoginResponseWithCredential extends CredentialResponse {
+    credential: string;
+}
```

Comment:
Duplicate in login page? This creates tech debt. There is two places that needs updating to keep in sync.